components/PopulationCalculation.tsx

```diff
@@ -0,0 +1,137 @@
+    Object.keys(currentPatients).forEach(id => {
+      const bundle = createPatientBundle(currentPatients[id].patient, currentPatients[id].resources);
+      patientBundles.push(bundle);
+    });
+
+    if (measureBundle.content) {
+      const measureReports = await Calculator.calculateMeasureReports(measureBundle.content, patientBundles, options);
+      return measureReports.results as fhir4.MeasureReport[];
+    } else return;
+  };
+
+  /**
+   * Wrapper function that calls calculateMeasureReports() and creates the DetailedMeasureReport that will be used to render
+   * the population results. Catches errors in fqm-execution that result from calculateMeasureReports().
+   */
+  const runCalculation = () => {
+    calculateMeasureReports()
+      .then(measureReports => {
+        if (measureReports) {
+          const patientLabels = createPatientLabels();
+          const labeledMeasureReports: DetailedMeasureReport[] = [];
+          measureReports.forEach((mr, i) => {
+            labeledMeasureReports.push({ label: patientLabels[i], report: mr as fhirJson.MeasureReport });
+          });
+          setMeasureReports(labeledMeasureReports);
+          setOpened(true);
+        }
+      })
+      .catch(e => {
+        if (e instanceof Error) {
+          showNotification({
+            icon: <IconAlertCircle />,
+            title: 'Calculation Error',
+            message: e.message,
+            color: 'red'
+          });
+        }
+      });
+  };
+
+  return (
+    <>
+      {Object.keys(currentPatients).length > 0 && measureBundle.content && (
+        <Center>
+          <Grid>
+            <Grid.Col span={12}>
+              <Button
+                data-testid="calculate-all-button"
+                aria-label="Calculate Population Results"
+                styles={{ root: { marginTop: 20 } }}
+                size="lg"
+                onClick={() => runCalculation()}
+              >
+                &nbsp;Calculate Population Results
+              </Button>
+              {measureReports.length > 0 && (
+                <>
+                  <div style={{ position: 'absolute', right: 0, padding: 5 }}>
+                    <Popover opened={opened} target={undefined}>
+                      <Group style={{ display: 'flex' }}>
+                        <h2>Population Results</h2>
+                        <Button
+                          leftIcon={<IconX size={18} />}
+                          aria-label="Close"
+                          onClick={() => setOpened(false)}
+                          style={{ marginLeft: 'auto' }}
+                          size="sm"
+                          color="red"
+                        >
+                          Close
+                        </Button>
+                      </Group>
+                      <div data-testid="results-table">
+                        <PopulationResultsViewer reports={measureReports} />
+                      </div>
+                    </Popover>
```

Comment:
I think this is a great design to start but I think we can make it even better even though this is a difficult UI challenge since there are so many things we want the user to be able to see at the same time. My first thought was to put the population result table to the right of the patient test cases so that under the buttons you see:
Resources | Patients | Population Results. I am not _quite_ sure how exactly this would look but I will make some sketches and share if that would be helpful! This is also where a second opinion from @mgramigna or @hossenlopp would be super helpful!